### PR TITLE
[DCOS-53557][COPS-4781] Expose Kibana web server through an nginx proxy.

### PR DESCRIPTION
**Pull request by mpereira**
This was done to address the issue surfaced by COPS-4781: accessing the Kibana service URL resulted in a redirect to an invalid URL. Namely:

```
GET /service/kibana/ => /service/kibana/service/kibana/app/kibana
```

The expected redirect should have been:

```
GET /service/kibana/ => /service/kibana/app/kibana
```

This happened because of a breaking change in Kibana 6.x which caused requests to the Kibana web server hitting the root path (`/`) resulted in an invalid redirect, depicted above.

This could be the place where this bug was introduced:
https://github.com/elastic/kibana/pull/21408/files#diff-230bb53cded88451213fc31d8a4b0774L103

The solution here was to add an nginx proxy in between Admin Router and the Kibana web server, that rewrites requests that would hit the Kibana web server on the root path to hit it on `/app/kibana` instead.

I also took the opportunity to extract the Kibana Marathon app command to an actual bash script, which is much more maintainable than the previous one-line `cmd` directive.

### CODE REVIEW

File: frameworks/elastic/universe-kibana/marathon.json.mustache [hunk 1]
```diff
@@ -4,15 +4,21 @@
...
+      "image": "{{resource.assets.container.docker.nginx}}"
+    }
+  },
+  "cmd": "./init.sh",
   "labels": {
     "DCOS_SERVICE_NAME": "{{service.name}}",
-    "DCOS_SERVICE_PORT_INDEX": "0",
-    "DCOS_SERVICE_SCHEME": "http"
+    "DCOS_SERVICE_PORT_INDEX": "1",
+    "DCOS_SERVICE_SCHEME": "http",
```
[the remark this answers is not in the document]
Comment (mpereira) [reply frameworks/elastic/universe-kibana/marathon.json.mustache]: The current TLS settings are for communications with the Elasticsearch cluster. We currently don't provide service configuration for running the Kibana web server in TLS. Fwiw, connections to it are proxied through Admin Router, which is already providing TLS encryption.

File: frameworks/elastic/universe-kibana/marathon.json.mustache [hunk 2]
```diff
@@ -4,15 +4,21 @@
...
   "instances": 1,
   "user": "{{service.user}}",
-  "cmd": "echo -e \"elasticsearch.url: $ELASTICSEARCH_URL\nelasticsearch.username: $KIBANA_USER\nelasticsearch.password: $KIBANA_PASSWORD\nserver.host: 0.0.0.0\nserver.port: $PORT_KIBANA\nserver.basePath: /service/$FRAMEWORK_NAME\nserver.rewriteBasePath: false\" > $MESOS_SANDBOX/kibana-$ELASTIC_VERSION-linux-x86_64/config/kibana.yml && echo -e \"\nxpack.security.encryptionKey: $MESOS_FRAMEWORK_ID\nxpack.reporting.encryptionKey: $MESOS_FRAMEWORK_ID\n\" >> $MESOS_SANDBOX/kibana-$ELASTIC_VERSION-linux-x86_64/config/kibana.yml && if [ \"$KIBANA_ELASTICSEARCH_TLS\" = true ]; then echo -e \"\nelasticsearch.ssl.certificateAuthorities: $MESOS_SANDBOX/.ssl/ca-bundle.crt\n\" >> $MESOS_SANDBOX/kibana-$ELASTIC_VERSION-linux-x86_64/config/kibana.yml; fi && env && $MESOS_SANDBOX/kibana-$ELASTIC_VERSION-linux-x86_64/bin/kibana",
+  "container": {
+    "type": "MESOS",
+    "docker": {
+      "image": "{{resource.assets.container.docker.nginx}}"
+    }
+  },
+  "cmd": "./init.sh",
```
[the remark this answers is not in the document]
Comment (mpereira) [reply frameworks/elastic/universe-kibana/marathon.json.mustache]: It was painful on the eyes...

**Comment by mpereira** [comment]
@samvantran thanks for taking a look. This change was partially covered by the existing tests, but I added two additional assertions: 17acf625f3ac91c86b6d97e5ed25bc381ba7f00a

Regarding the possibility of not needing the nginx proxy because of a possible fix for Kibana on future versions, it might still make sense to keep it because it makes it possible for us to fix the issue detailed in this ticket: https://jira.mesosphere.com/browse/DCOS-46801

File: frameworks/elastic/universe-kibana/marathon.json.mustache (hunk 1 above)
[the remark this answers is not in the document]
Comment (samvantran) [reply frameworks/elastic/universe-kibana/marathon.json.mustache]: fair point 👍

**Comment by samvantran** [comment]
Thanks for additional assertions. @rishabh96b please take a look too !

**Comment by mpereira** [comment]
The Elastic strict build passed, but the permissive build failed failed on a known flake: https://teamcity.mesosphere.io/viewLog.html?buildId=1878143&buildTypeId=DataServices_Elastic_IntegrationTests_DCOS_master_Permissive_PR&guest=1

Merging this for now.

**Pull request merged**